dali/kernels/test/mat2tensor.h

```diff
@@ -16,43 +16,51 @@
 #ifndef DALI_KERNELS_TEST_MAT2TENSOR_H_
 #define DALI_KERNELS_TEST_MAT2TENSOR_H_
 
-#include <gtest/gtest.h>
 #include <opencv2/core.hpp>
+#include <stdexcept>
 #include "dali/kernels/tensor_view.h"
 
 namespace dali {
 namespace kernels {
 
-template <typename T, int ndim = 3>
-TensorView<StorageCPU, const T, ndim> view_as_tensor(const cv::Mat &mat) {
+template <int ndim>
+TensorShape<ndim> tensor_shape(const cv::Mat &mat) {
   TensorShape<ndim> shape;
   if (ndim != DynamicDimensions) {
-    ASSERT_EQ(ndim, mat.size.dims() + 1), TensorView<StorageCPU, const T, ndim>();
+    // require extra dimension for channels, if #channels > 1
+    if (ndim != mat.size.dims() + 1 && !(ndim == mat.size.dims() && mat.channels() == 1))
+      throw std::logic_error("Invalid number of dimensions");
   } else {
     shape.resize(mat.size.dims() + 1);
   }
-  int depth = cv::DataDepth<T>::value;
-  ASSERT_EQ(mat.depth(), depth), TensorView<StorageCPU, const T, ndim>();
-  for (int i = 0; i < mat.size.dims(); i++)
+  for (int i = 0; i < shape.size(); i++)
     shape[i] = mat.size[i];
-  shape[mat.size.dims()] = mat.channels();
-  return { mat.data, shape };
+  if (shape.size() > mat.size.dims())
+    shape[mat.size.dims()] = mat.channels();
+  return shape;
+}
+
+template <typename T>
+void enforce_type(const cv::Mat &mat) {
+  using U = typename std::remove_const<T>::type;
+  int depth = cv::DataDepth<U>::value;
+  if (depth != mat.depth())
+    throw std::logic_error("Invalid matrix data type");
```

Comment:
wondering why logic error and not runtime error, for instance

Reply:
Also, currently we/Python are catching only runtime_error

Reply:
I found it similar to `std::out_of_range` which inherits from `std::logic_error`.